test/unit/utility/views/zip_test.cpp

```diff
@@ -5,20 +5,189 @@
 // shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
 // -----------------------------------------------------------------------------------------------------
 
-#include <algorithm>
-#include <deque>
-#include <iostream>
-#include <list>
-#include <seqan3/std/ranges>
-#include <string>
-#include <vector>
-
 #include <gtest/gtest.h>
 
+#include <forward_list>
+
+#include <seqan3/alphabet/nucleotide/dna4.hpp>
+#include <seqan3/utility/views/repeat.hpp>
+#include <seqan3/utility/views/chunk.hpp>
 #include <seqan3/utility/views/zip.hpp>
 
```

Comment:
Please add the iterator test template. It has extensive tests for iterators.

```suggestion
#include "../../../range/iterator_test_template.hpp"

using range_t = std::vector<int>;
using other_range_t = std::vector<std::string>;
using zip_view_t = decltype(seqan3::views::zip(range, other_range));
using zip_iterator_t = typename zip_view_t::iterator;

template <>
struct iterator_fixture<zip_iterator_t> : public ::testing::Test
{
    using iterator_tag = std::random_access_iterator_tag;
    static constexpr bool const_iterable = true; // I guess?

    range_t range{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
    other_range_t other_range{"AA", "BBB", "CC", "DDD"};

    std::vector<seqan3::common_pair<int, std::string>> expected_range{{0, "AA"}, {1, "BBB"}, {2, "CC"}, {3, "DDD"}};

    zip_view_t test_range{seqan3::views::zip(range, other_range)};

    void SetUp()
    {
        // probably not needed?
    }

    // probably not needed as common_pair is == comparable
    // template <typename value_type1, typename value_type2>
    // static void expect_eq(value_type1 && v1, value_type2 && v2)
    // {
    // }
};

INSTANTIATE_TYPED_TEST_SUITE_P(zip_iterator_test, iterator_fixture, zip_iterator_t, );

```

Probably won't compile right away but will save you at least some time.